### ISSUE TITLE
Index out of range exception on console startup

### ISSUE THREAD

**Issue by jennablackwell** [eric-king/BingoEngine#1]
Thank you for sharing this project!  Just a heads up when I try to run the console app in it, I do receive an index out of range exception..

<img width="823" alt="image" src="https://user-images.githubusercontent.com/14877652/225342545-925573c7-84b1-4ee4-b868-9072d34be344.png">

**Comment by eric-king** [comment]
Does this happen with every pattern you choose, or a specific pattern? Did you alter any of the pattern classes?

**Comment by jennablackwell** [comment]
Hi Eric,
I didn't modify anything - just tried to run the console app after
downloading the project.  Were any arguments supposed to be provided when
running?

**Comment by eric-king** [comment]
No, it should run out of the box. Based on where the error is happening, I'm guessing that maybe the `pictogram.Split(Environment.NewLine)` is the issue. Perhaps the roundtrip from Windows -> GitHub -> Windows is messing up the encoding or something. I haven't been able to recreate the issue yet, so I'll probably try to clone everything fresh and see if that helps.

**Comment by eric-king** [comment]
Hmm... I did a fresh clone, built it, and ran it with no problems. I'm running it in VS2022 on Windows. Is your environment different?

**Comment by jennablackwell** [comment]
Weird - yes VS 2022 on Windows 11.

**Comment by jeffpowerscr** [comment]
im having the same problem? index out of range... vs 2022 windows 11

**Comment by jeffpowerscr** [comment]
When i cloned the project into VS, i had no issue.. when i downloaded the ZIP file, extracted, and ran the solution file, i got the index error when starting up.

**Comment by eric-king** [comment]
> When i cloned the project into VS, i had no issue.. when i downloaded the ZIP file, extracted, and ran the solution file, i got the index error when starting up.

That's pretty strange. I don't know what would be different between the zip file and cloning.

**Comment by pr0ctor** [comment]
@eric-king So I think I found the source of the issue but not 100% why it happens depending on if the project is cloned vs downloaded through a .zip file. I am also using Visual Studio 2022 on Windows 10 and ran into this exception after having downloaded the source code as a .zip file.

The issue seems to stem from the `Environment.NewLine` value and an inconsistency between newlines in the source Pictograms and in the user's `Environment`.

Here in the screenshot when loading the first pattern Pictogram `BlackoutPattern` we can see that after hitting line 40, the `pictogram.Split(Environment.NewLine)` yielded an array that was technically split, but not in the format that we were wanting.

![image](https://github.com/user-attachments/assets/7b120211-8f78-44b2-8dfe-0bfa2394234e)

This is because on Windows, the `Environment.NewLine` value resolved to `\r\n` (`carriage return` then `newline`) whereas on unix based systems the value is just `\n`. We can then see that the values in the Pattern only contain `\n` so for some reason the `\r` was stripped out during some part of the git process.

Investigating the whitespace characters in Notepad++ confirms the lack of `\r` in the source code in this case:

![image](https://github.com/user-attachments/assets/a4b065e3-33f0-4170-ba74-2fe9d8d828f4)

When I changed the `pictogram.Split(Environment.NewLine)` to just `pictogram.Split("\n")` the code worked perfectly fine in my case. I think that there can be a few solutions to this issue depending on the approach that you're wanting to take. One way could potentially be replacing the `Environment.NewLine` with an adjusted 
`pictogram.Split(new string[] { "\r\n", "\r", "\n" }, StringSplitOptions.TrimEntries)`. 
This will split on any of the given values and trim the resulting string.

I wasn't able to publish a branch with this change or open a PR but this should fix the issue.
